File: 0020-valid-parentheses/0020-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> stc = new Stack<>();
        for (char ch : s.toCharArray()){
            if(ch == '(' || ch == '[' || ch == '{')
                stc.push(ch);
            else{
                if (stc.empty()) return false;
                if ((stc.peek() == '(' && ch != ')') ||
                    (stc.peek() == '[' && ch != ']') || 
                    (stc.peek() == '{' && ch != '}'))
                        return false;
                stc.pop();
            }
        }
        return stc.isEmpty();
    }
}

File: 0020-valid-parentheses/0020-valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> stc;
        int len = s.size();
        if (len <= 1) return false;
        for(char ch:s){
            if(ch == '(' || ch == '[' || ch == '{')
                stc.push(ch);
            else{
                if (stc.empty()) return false;
                if ((stc.top() == '(' && ch != ')') ||
                    (stc.top() == '[' && ch != ']') || 
                    (stc.top() == '{' && ch != '}'))
                        return false;
                stc.pop();
            }
        }
        return stc.empty();
    }
};